File: example-webapp/src/main/java/vip/yazilim/libs/springvip/template/config/springvip/CustomRestResponseBuilder.java
package vip.yazilim.libs.springvip.template.config.springvip;

import org.jetbrains.annotations.NotNull;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Component;
import vip.yazilim.libs.springvip.config.bean.IRestResponseBuilder;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.util.LinkedHashMap;
import java.util.Map;

/**
 * @author maemresen - maemresen@yazilim.vip
 * 21.08.2020
 */
@Component
public class CustomRestResponseBuilder implements IRestResponseBuilder {
    @NotNull
    @Override
    public Object generateRestResponse(@NotNull Object responseBody, @NotNull HttpStatus httpStatus, @NotNull HttpServletRequest request, @NotNull HttpServletResponse response) {
        Map<String, Object> map = new LinkedHashMap<>();
        map.put("info", "my custom response model");
        map.put("my-data", responseBody);
        return map;
    }
}

File: src/main/kotlin/vip/yazilim/springvip/templates/kotlin/config/CustomRestResponseBuilder.kt
package vip.yazilim.springvip.templates.kotlin.config

import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpStatus
import org.springframework.stereotype.Component
import vip.yazilim.libs.springvip.config.bean.IRestResponseBuilder
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * Cust
 * @author maemresen - maemresen@yazilim.vip
 * 12/31/2020
 */
@Component
class CustomRestResponseBuilder : IRestResponseBuilder {
    override fun generateRestResponse(
        responseBody: Any,
        httpStatus: HttpStatus,
        request: HttpServletRequest,
        response: HttpServletResponse
    ): Any {
        return mapOf(
            Pair("my-data", responseBody),
            Pair("additional-data-1", "value"),
            Pair("additional-data-2", true)
        )
    }

}